relay-config/src/config.rs

```diff
@@ -784,9 +882,30 @@ pub struct Processing {
     pub max_session_secs_in_past: u32,
     /// Kafka producer configurations.
     pub kafka_config: Vec<KafkaConfigParam>,
+    /// Additional kafka producer configurations.
+    ///
+    /// The `kafka_config` is the default producer configuration used for all topics. A secondary
+    /// kafka config can be referenced in `topics:` like this:
+    ///
+    /// ```yaml
+    /// secondary_kafka_configs:
+    ///   mycustomcluster:
+    ///     - name: 'bootstrap.servers'
+    ///       value: 'sentry_kafka_metrics:9093'
+    ///
+    /// topics:
+    ///   transactions: ingest-transactions
+    ///   metrics:
+    ///     topic_name: ingest-metrics
+    ///     kafka_config_name: mycustomcluster
+    /// ```
+    ///
+    /// Then metrics will be produced to an entirely different Kafka cluster.
+    #[serde(default)]
+    pub secondary_kafka_configs: BTreeMap<String, Vec<KafkaConfigParam>>,
     /// Kafka topic names.
     #[serde(default)]
-    pub topics: TopicNames,
+    pub topics: TopicMap<TopicAssignment>,
```

Comment:
Would add a type alias for `TopicMap<TopicAssignment>`, and maybe rename the generic type to something more that indicates its generic structure.